crates/miden-tx/src/executor/mod.rs

```diff
@@ -130,26 +106,38 @@ impl TransactionExecutor {
         &self,
         account_id: AccountId,
         block_ref: BlockNumber,
-        notes: &[NoteId],
+        notes: InputNotes<InputNote>,
         tx_args: TransactionArgs,
     ) -> Result<ExecutedTransaction, TransactionExecutorError> {
-        let tx_inputs =
-            maybe_await!(self.data_store.get_transaction_inputs(account_id, block_ref, notes))
+        let mut ref_blocks: BTreeSet<BlockNumber> = notes
+            .iter()
+            .filter_map(InputNote::location)
+            .map(NoteLocation::block_num)
+            .collect();
+        ref_blocks.insert(block_ref);
+
+        let (account, seed, header, mmr) =
+            maybe_await!(self.data_store.get_transaction_inputs(account_id, ref_blocks))
                 .map_err(TransactionExecutorError::FetchTransactionInputsFailed)?;
 
+        let tx_inputs = TransactionInputs::new(account, seed, header, mmr, notes)
+            .map_err(TransactionExecutorError::InvalidTransactionInputs)?;
+
         let (stack_inputs, advice_inputs) =
             TransactionKernel::prepare_inputs(&tx_inputs, &tx_args, None);
-        let advice_recorder: RecAdviceProvider = advice_inputs.into();
 
-        // load note script MAST into the MAST store
-        self.mast_store.load_transaction_code(&tx_inputs, &tx_args);
+        let advice_recorder: RecAdviceProvider = advice_inputs.into();
 
         let mut host = TransactionHost::new(
             tx_inputs.account().into(),
             advice_recorder,
-            self.mast_store.clone(),
+            self.data_store.clone(),
             self.authenticator.clone(),
-            self.account_codes.iter().map(|code| code.commitment()).collect(),
+            tx_args
+                .foreign_accounts()
+                .iter()
+                .map(|acc| acc.account_code().commitment())
+                .collect(),
```

Comment:
nit: I'd probably move this out into a separate variable - e.g., `foreign_account_code_commitments`. Or maybe even this could be a method on `TransactionArgs`.

Reply:
Added a method